lib/rows.js

```diff
@@ -343,4 +343,52 @@ Rows.prototype.explain = function explainRows(builtPlan, format, queryType) {
   return responseSelector.result();
 };
 
+/**
+ * Generates the view of an execution plan as a promise.
+ * @method rows#generateView
+ * @since 2.7.0
+ * @param {object} builtPlan - A {@link planBuilder} object or a built
+ * plan as an XML object.
+ * @param {string} [schema] - The schema of the query.
+ * @param {string} [view] - View of the query.
+ * @param {string} [queryType] - The type of a string query submitted for execution.
+ * @returns {Promise} A promise whose success callback receives the explanation as an XML.
+ */
+Rows.prototype.generateView = function generateViewRows(builtPlan, schema, view, queryType) {
+  const connectionParams = this.client.getConnectionParams();
+  const requestOptions = mlutil.copyProperties(connectionParams);
+  requestOptions.method = 'POST';
+
+  if(schema == null || schema.length ===0) {
+    throw new Error('schema cannot be null or empty');
+  }
+  if(view == null || view.length ===0) {
+    throw new Error('view cannot be null or empty');
+  }
+
+  let acceptHeader = 'application/xml';
+
+  const contentTypeHeader = queryContentType(builtPlan, queryType);
+
+  requestOptions.headers = {
+    'Content-Type': contentTypeHeader,
+    'Accept': acceptHeader
+  };
+
+  requestOptions.path = '/v1/rows?output=generateView';
+
+  const operation = new Operation(
+      'explain rows', this.client, requestOptions, 'single', 'single');
+  operation.validStatusCodes  = [200, 404];
```

Comment:
404 is a valid code when retrieving rows because the query might not match any data.

In this case, it would be an error if the query couldn't be generated -- also in the explain case, which should only allow 200.